pkg/apis/node/validation/validation.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	genericvalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/apis/node"
+)
+
+// ValidateRuntimeClass validates the RuntimeClass
+func ValidateRuntimeClass(rc *node.RuntimeClass) field.ErrorList {
+	allErrs := genericvalidation.ValidateObjectMeta(&rc.ObjectMeta, false, genericvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))
+	allErrs = append(allErrs, ValidateRuntimeClassSpec(&rc.Spec, field.NewPath("spec"))...)
+	return allErrs
+}
+
+// ValidateRuntimeClassSpec validates the sink spec for audit
+func ValidateRuntimeClassSpec(spec *node.RuntimeClassSpec, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if spec.RuntimeHandler != nil && *spec.RuntimeHandler != "" {
+		for _, msg := range genericvalidation.NameIsDNSSubdomain(*spec.RuntimeHandler, false) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("runtimeHandler"), *spec.RuntimeHandler, msg))
+		}
+	}
+	return allErrs
+}
+
+// ValidateRuntimeClassUpdate validates an update to the object
+func ValidateRuntimeClassUpdate(new, old *node.RuntimeClass) field.ErrorList {
+	allErrs := genericvalidation.ValidateObjectMetaUpdate(&new.ObjectMeta, &old.ObjectMeta, field.NewPath("metadata"))
+
+	// All RuntimeClassSpec fields are immutable.
```

Comment:
Assuming one of the future use cases is to explicitly specify resource overhead of containerizers, I can imagine such an overhead changing across different versions of containernerizers. Does this comment hold good or is the use case I'm referring to no longer in scope?

Reply:
Yes, but all RuntimeClassSpec fields are **currently** immutable. I think we can just update this when we add a mutable field?

Reply:
Ok.